app/assets/javascripts/listings/ListingService.js.coffee

```diff
@@ -81,6 +81,24 @@ ListingService = ($http, $localStorage, $modal, $q, $state) ->
       minMax = [min, max]
     return minMax
 
+  Service.hasPreference = (preference) ->
+    hasPreference = null
```

Comment:
not that it will have any real difference in outcome, but this should probably default to `false`, that way `hasPreference` either returns `true` or `false`, rather than `true` or `null`.